plugins/repository-azure/src/test/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceMock.java

```diff
@@ -94,8 +94,15 @@ public OutputStream getOutputStream(String account, LocationMode mode, String co
     public Map<String, BlobMetaData> listBlobsByPrefix(String account, LocationMode mode, String container, String keyPath, String prefix) {
         MapBuilder<String, BlobMetaData> blobsBuilder = MapBuilder.newMapBuilder();
         for (String blobName : blobs.keySet()) {
-            if (startsWithIgnoreCase(blobName, prefix)) {
-                blobsBuilder.put(blobName, new PlainBlobMetaData(blobName, blobs.get(blobName).size()));
+            final String checkBlob;
+            if (keyPath != null) {
+                // strip off key path from the beginning of the blob name
+                checkBlob = blobName.replace(keyPath, "");
```

Comment:
@ywelsch to further elaborate, the lack of taking the `keyPath` into account when listing blobs was a bug in`AzureStorageServiceMock`.  The reason this never manifested before is the `listBlobsByPrefix` was only called from `BlobStoreRepository#snapshots()`, which in turn was only called during `initializeSnapshot` and `finalizeSnapshot` to get the current list of snapshots in the index file to add to it.  This means the snapshot file did not have the correct entries, but it didn't matter because it was never used anyway (because `listBlobsByPrefix` was always invoked instead).  However, because now we are also using the `BlobStoreRepository#snapshots()` method in `resolveSnapshotNames`, it has manifested this issue in the `AzureStorageServiceMock`.  Hence the change above.